src/Mono.Android/Android.Runtime/JavaObject.cs

```diff
@@ -10,56 +10,54 @@ public static IntPtr GetHandle (object obj)
 			if (obj == null)
 				return IntPtr.Zero;
 
-			Type type = obj.GetType ();
-			if (type == typeof (bool))
-				return new Java.Lang.Boolean ((bool)obj).Handle;
-			else if (type == typeof (sbyte))
-				return new Java.Lang.Byte ((sbyte)obj).Handle;
-			else if (type == typeof (char))
-				return new Java.Lang.Character ((char)obj).Handle;
-			else if (type == typeof (short))
-				return new Java.Lang.Short ((short)obj).Handle;
-			else if (type == typeof (int))
-				return new Java.Lang.Integer ((int)obj).Handle;
-			else if (type == typeof (long))
-				return new Java.Lang.Long ((long)obj).Handle;
-			else if (type == typeof (float))
-				return new Java.Lang.Float ((float)obj).Handle;
-			else if (type == typeof (double))
-				return new Java.Lang.Double ((double)obj).Handle;
-			else if (type == typeof (string))
-				return JNIEnv.NewString ((string)obj);
-			else if (typeof (IJavaObject).IsAssignableFrom (type))
+			if (obj is bool bool_obj)
+				return Java.Lang.Boolean.ValueOf (bool_obj).Handle;
+			else if (obj is sbyte sbyte_obj)
+				return Java.Lang.Byte.ValueOf (sbyte_obj).Handle;
+			else if (obj is char char_obj)
+				return Java.Lang.Character.ValueOf (char_obj).Handle;
+			else if (obj is short short_obj)
+				return Java.Lang.Short.ValueOf (short_obj)!.Handle;
+			else if (obj is int int_obj)
+				return Java.Lang.Integer.ValueOf (int_obj).Handle;
+			else if (obj is long long_obj)
+				return Java.Lang.Long.ValueOf (long_obj).Handle;
+			else if (obj is float float_obj)
+				return Java.Lang.Float.ValueOf (float_obj).Handle;
+			else if (obj is double double_obj)
+				return Java.Lang.Double.ValueOf (double_obj).Handle;
+			else if (obj is string string_obj)
+				return JNIEnv.NewString (string_obj);
+			else if (typeof (IJavaObject).IsAssignableFrom (obj.GetType ()))
 				return ((IJavaObject)obj).Handle;
 			else
 				return new JavaObject (obj).Handle;
 		}
 
 		public static object? GetObject (IntPtr handle, JniHandleOwnership transfer)
 		{
-			Java.Lang.Object? jlo = Java.Lang.Object.GetObject (handle, transfer) as Java.Lang.Object;
-			if (jlo == null)
+			if (Java.Lang.Object.GetObject (handle, transfer) is not Java.Lang.Object jlo)
```

Comment:
Similarly, what calls `JavaObject.GetObject()`?   `git grep JavaObject.GetObject` only finds:

```
src/Mono.Android/Android.Runtime/JNIEnv.cs:                     var javaException = JavaObject.GetObject<Java.Lang.Throwable> (env, javaExceptionPtr, JniHandleOwnership.DoNotTransfer)!;
```

which is a *generic* method invocation, and thus can't be *this* method.  Is this method actually used anymore?  Can we just remove it?